astropy/io/fits/connect.py

```diff
@@ -448,7 +448,7 @@ def write_table_fits(input, output, overwrite=False, append=False):
     table_hdu = table_to_hdu(input, character_as_bytes=True)
 
     # Check if output file already exists
-    if isinstance(output, str) and os.path.exists(output):
+    if os.path.exists(output) and os.path.isfile(output):
```

Comment:
this is clearly a better approach than stacking `isinstance` calls. However, the fact that this function now explicitly accepts `Path` should be be reflected in the docstring too